src/ngraph_builder.cc

```diff
@@ -374,6 +374,98 @@ static tf::Status TranslateAvgPoolOp(const tf::Node* op,
   return tf::Status::OK();
 }
 
+static tf::Status TranslateAvgPoolGradOp(const tf::Node* op, 
+                                     Builder::OpMap& ng_op_map) { 
+  TF_RETURN_IF_ERROR(ValidateInputCount(op, 2));
+      
+  shared_ptr<ng::Node> ng_grad;
+  TF_RETURN_IF_ERROR(GetInputNode(ng_op_map, op, 1, &ng_grad));
+
+  std::vector<tf::int32> tf_orig_input_shape_vec;
+  TF_RETURN_IF_ERROR(tf::GetNodeAttr(
+      op->attrs(), "_ngraph_avgpoolgrad_static_input_shape", &tf_orig_input_shape_vec));
+
+  std::vector<tf::int32> tf_strides;
+  std::vector<tf::int32> tf_ksize;
+  std::string tf_padding_type;
+  std::string tf_data_format;
+  TF_RETURN_IF_ERROR(tf::GetNodeAttr(op->attrs(), "strides", &tf_strides));
+  TF_RETURN_IF_ERROR(tf::GetNodeAttr(op->attrs(), "ksize", &tf_ksize));
+  TF_RETURN_IF_ERROR(
+      tf::GetNodeAttr(op->attrs(), "padding", &tf_padding_type));
+  TF_RETURN_IF_ERROR(
+      tf::GetNodeAttr(op->attrs(), "data_format", &tf_data_format));
+
+  if (tf_data_format != "NHWC" && tf_data_format != "NCHW") {
+    return tf::errors::InvalidArgument(
+        "AvgPoolGrad data format is neither NHWC nor NCHW");
+  }
+
+  bool is_nhwc = (tf_data_format == "NHWC");
+
+  NGRAPH_VLOG(3) << ng::join(tf_strides);
+  NGRAPH_VLOG(3) << ng::join(tf_ksize);
+  NGRAPH_VLOG(3) << tf_padding_type;
+  NGRAPH_VLOG(3) << tf_data_format;
+
+  ng::Shape ng_forward_arg_shape;
+  ng::Strides ng_strides(2);
+  ng::Shape ng_image_shape(2);
+  ng::Shape ng_kernel_shape(2);
+
+  if (tf_orig_input_shape_vec.size() != 4) {
+    return tf::errors::InvalidArgument(
+        "Tensorflow requires input tensor shape being 4", tf_orig_input_shape_vec.size());
+  }
+  for (int i = 0; i < tf_orig_input_shape_vec.size(); i++) {
+    ng_forward_arg_shape.push_back(tf_orig_input_shape_vec[i]);
+  }
+
+  if (is_nhwc) {
```

Comment:
Can we use `BatchedOpParamToNGraph` for this?

Reply:
:+1: this is also from refactor changes in `master`

Reply:
Added a new helper function in `ngraph_conversions.h`for this case.